pkgs/development/libraries/onnxruntime/default.nix

```diff
@@ -0,0 +1,107 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchpatch
+, runCommand
+, patchutils
+, pkgconfig
+, glibcLocales
+, cmake
+, python3
+, libpng
+, zlib
+, eigen
+, protobuf
+, nsync
+, flatbuffers
+, howard-hinnant-date
+, re2
+, nlohmann_json
+, boost
+}:
+
+stdenv.mkDerivation rec {
+  pname = "onnxruntime";
+  version = "1.10.0";
+
+  src = fetchFromGitHub {
+    owner = "microsoft";
+    repo = "onnxruntime";
+    rev = "v${version}";
+    sha256 = "sha256-V56JOMT74oxB3VQ0dKH2Nc6d3OmjPGqBLZ3yDcrcOKA=";
+    # TODO: Use Nix provide only the source submodules that are needed.
+    fetchSubmodules = true;
+    # Remove unicode file names which leads to different checksums on HFS+
+    # vs. other filesystems because of unicode normalisation.
+    postFetch = ''
+      rm -rf $out/winml/test/collateral/models/UnicodePath/
+    '';
+  };
```

Comment:
would be nice if we could slim this down a bit.... but it would probably require significant upstream effort

```
$ realpath ./result
/nix/store/pygayskn6aap3473n4j0y7ikzl0klf2c-source
$ du -hd0 ./result/
377M	./result/
```

Reply:
Is it possible to delete the submodules we are not interested in in `postFetch`?

Reply:
yes, but it will affect the sha, so you have to re-compute that

Reply:
Originally I tried to find a way to change `.gitmodules` during the clone but the Nix wrapper won't allow that. Sucks because maintaining a fork of just that file would actually make it possible, but I really don't want to do that.

Reply:
Turns out there were only 3 required repositories. I've written some Nix to fetch them.

Reply:
That's very nice!